### PR TITLE
fix(NcCheckboxRadioSwitch): fix shift+click on firefox

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix https://github.com/nextcloud/server/issues/42202
Ref https://bugzilla.mozilla.org/show_bug.cgi?id=559506

I'm not sure, if anyone else have a better idea, please suggest it! :pray: :smiling_face_with_tear:

### CODE REVIEW

**Review by emoral435 — APPROVED** [review]
I actually primarily use Firefox, and am working on an issue that needs this resolved, so this is perfect. LGTM, works on my dev env, and keeps the aria label logic clean :)

**Comment by ShGKme** [comment]
> Ref https://bugzilla.mozilla.org/show_bug.cgi?id=151142

Isn't it this bug?
https://bugzilla.mozilla.org/show_bug.cgi?id=559506

File: src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue [hunk 1]
```diff
@@ -276,7 +276,7 @@ export default {
 		:type="isButtonType ? 'button' : null"
 		v-on="isButtonType ? listeners : null">
 		<input v-if="!isButtonType"
-			:id="id"
+			:aria-labelledby="!isButtonType && !ariaLabel ? id : null"
```
Comment (ShGKme) [inline src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue:279]: Isn't it a breaking change?

This component has a prop `id` which still has description `Unique id attribute of the input`.

But not an instance with `id` set won't render an `input` with this `id` but a `span`.

Example of breaking change:
```html
<!-- label doesn't work anymore -->
<label for="my-id">Custom label outside</label>

<NcCheckboxRadioSwitch id="my-id" />

<script>
// Doesn't work
document.getElementById('my-id').focus()
</script>
```

Maybe examples are edge cases, but they are supposed to work according to the current description `id attribute of the input`

And the same with `NcCheckboxContent` where the meaning of `id` prop has changed.

At least I'd change the prop descriptions.

---

As an alternative, I may propose to not use `id` from prop here, but generate a new local `id` and pass it to the `NcCheckboxContent` as a new prop.

**Comment by skjnldsv** [comment]
> > Ref [bugzilla.mozilla.org/show_bug.cgi?id=151142](https://bugzilla.mozilla.org/show_bug.cgi?id=151142)
> 
> Isn't it this bug? [bugzilla.mozilla.org/show_bug.cgi?id=559506](https://bugzilla.mozilla.org/show_bug.cgi?id=559506)

Wrong paste, my bad!

File: src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue (hunk 1 above)
Comment (skjnldsv) [reply src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue:279]: Ah, right, I used the wrong id maybe, I'll have a second look this morning! Thanks @ShGKme  :)

Comment (skjnldsv) [reply src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue:279]: image

Here we go

**Review by ShGKme — APPROVED** [review]

**Comment by ShGKme** [comment]
But I'd add a comment in the code with a link to the bug, because it seems non-obvious why a native element is not used here.

**Pull request merged**